spring-cloud-starter-stream-source-mongodb/src/test/java/org/springframework/cloud/stream/app/mongodb/source/MongodbSourceApplicationTests.java

```diff
@@ -16,121 +16,176 @@
-			Message<?> received = messageCollector.forChannel(source.output()).poll(2, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.notNullValue());
-			assertThat((String)received.getPayload(), CoreMatchers.containsString("hola"));
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(2, TimeUnit.SECONDS);
+			assertThat(received, notNullValue());
+			assertThat((String) received.getPayload(), containsString("hola"));
 		}
+
 	}
 
-	@IntegrationTest(value = {"mongodb.collection=testing", "mongodb.query={ 'greeting': 'bogus' }", "trigger.fixedDelay=1"})
+	@TestPropertySource(properties = {
+			"mongodb.query={ 'greeting': 'bogus' }",
+			"trigger.fixedDelay=1" })
 	public static class InvalidQueryTests extends MongodbSourceApplicationTests {
+
 		@Test
 		public void test() throws InterruptedException {
-			Message<?> received = messageCollector.forChannel(source.output()).poll(2, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.nullValue());
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(2, TimeUnit.SECONDS);
+			assertThat(received, nullValue());
 		}
+
 	}
 
-	@IntegrationTest(value = {"mongodb.collection=testing", "trigger.fixedDelay=1", "mongodb.split=false"})
+	@TestPropertySource(properties = {
+			"trigger.fixedDelay=1",
+			"mongodb.split=false" })
 	public static class NoSplitTests extends MongodbSourceApplicationTests {
+
 		@Test
 		public void test() throws InterruptedException {
-			Message<?> received = messageCollector.forChannel(source.output()).poll(2, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.notNullValue());
-			assertThat(received.getPayload(), Matchers.instanceOf(List.class));
-			assertThat(received.getPayload().toString(), CoreMatchers.containsString("hola"));
-			assertThat(received.getPayload().toString(), CoreMatchers.containsString("hello"));
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(2, TimeUnit.SECONDS);
+			assertThat(received, notNullValue());
+			assertThat(received.getPayload(), instanceOf(List.class));
+			assertThat(received.getPayload().toString(), containsString("hola"));
+			assertThat(received.getPayload().toString(), containsString("hello"));
 		}
+
 	}
 
-	@IntegrationTest(value = {"mongodb.collection=testing", "trigger.fixedDelay=100"})
+	@TestPropertySource(properties = "trigger.fixedDelay=100")
 	public static class MongoTriggerTests extends MongodbSourceApplicationTests {
+
 		@Test
 		public void test() throws InterruptedException {
-			Message<?> received = messageCollector.forChannel(source.output()).poll(1, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.nullValue());
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(1, TimeUnit.SECONDS);
+			assertThat(received, nullValue());
```

Comment:
This test seems to be bogus; it's the same as `DefaultTests` but with a long trigger - it seems to be just testing that the trigger doesn't fire within 1 second.